app/models/study_file.rb

```diff
@@ -896,6 +896,23 @@ def remove_local_copy
     Rails.logger.info "Removal of local copy of #{self.upload_file_name} complete"
   end
 
+  # check if this file can be deleted "safely"; e.g. not being used in any running parse jobs
+  # most files just need to check if they are still parsing; cluster/metadata files need to check for subsampling
+  def can_delete_safely?
```

Comment:
this logic is juuuust complex enough that it might be worth a model-level test.  E.g. just create a study file object with the file type and is_parsing  set/unset and manually create a clustergroup object for it with is_subsampling set/unset.  (this may not be possible depending on what kind of hooks/model validations have to pass, but if you can do it in <10mins, I think it's worth it)

Reply:
I initially wasn't going to because I thought that it would involve another long-running integration test that would add several minutes to our CI runs, but on reflection your suggestion will be more than sufficient.  I'll put something together.

Reply:
5e47004